lib/vector-core/src/event/metric/tags.rs

```diff
@@ -71,24 +70,28 @@ impl TagValueSet {
     }
 
     #[cfg(test)]
-    fn contains(&self, value: &str) -> bool {
+    fn contains<T>(&self, value: &T) -> bool
+    where
+        T: Eq + Hash + PartialEq<Option<String>>,
+        TagValue: Borrow<T>,
+    {
         match self {
             Self::Single(None) => false,
-            Self::Single(Some(tag)) => tag == value,
+            Self::Single(Some(tag)) => value == tag,
```

Comment:
Why the order swap?

Reply:
Because `value` is the one that is constrained to implement `PartialEq` and not the other way around.

Reply:
Ahhhh, gotcha. 👍🏻